File: 27-Special Array With X Elements Greater Than or Equal X/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int specialArray(vector<int>& nums) {
        int x = nums.size();
        int cnt;

        while (x != 0) {
            cnt = 0;
            for (int n : nums) {
                if (n >= x) cnt++;
            }

            if (cnt == x) return x;

            x--;
        }

        return -1;
    }
};

File: 27-Special Array With X Elements Greater Than or Equal X/solution.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        x = len(nums)
        counter = 0
        while x != 0:
            counter = 0
            for num in nums:
                if num >= x:
                    counter += 1
            if counter == x:
                return x
            x -= 1
            
        return -1